IA-3364 user history front end

quang-le opened the pull request:
Explain what problem this PR is resolving

Related JIRA tickets : IA-3364

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc


## Changes

- Add sub menu to "users" menu entry.
- Use `USERS_ADMIN" permission to control access
- List view:
    - Call `userlogs` api
    - Populate table
    - Filters:
        - users
        - modification date
        - location: filters on past and new org units
- Details view:
    - Use table `subComponent` to expand the view under the current row
    - call `logs` api to get data
    - Duplicate all logs related components and adapt them for users   

## How to test

- Go to users > management
- create a user
- update a user
- update multiple users by selecting them via the table. IMPORTANT: you need an active worker for this part
- create multiple users via csv

Go to users > history:
- All creations and updates should appear in the table
- Test out all the filters
- Expand logs to compare before and after views.



## Print screen / video

Uploading Screen Recording 2024-09-09 at 17.08.00.mov…



## Notes

Merges into https://github.com/BLSQ/iaso/pull/1614

quang-le committed:
IA-3364: add user history entry in menu

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: display table

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: table+filters

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: add front-end for list view

quang-le committed:
Merge branch 'IA-3364_user_history_front-end' of https://github.com/BLSQ/iaso into IA-3364_user_history_front-end

quang-le committed:
IA-3364: small table fixes

- fix default pageSize
- remove sort on columns with list values

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: update Logs components to decouple from org units

quang-le committed:
IA-3364: WIP: add action column to Users log table

quang-le committed:
IA-3364: add sub component to users history table

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: WIp history log details

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: use generic LogDetails for user logs

quang-le committed:
IA-3364: make custom log for user history (front-end)

- Add UserHistoryLogDetails, UserLogCompare, UserLogValueWithErrorBoundary, UserLogValue
- All are derived from the existing Log components which are specific to org units and repurposed for users
- Add translations

quang-le committed:
IA-3364: update menu

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: handle changed fields column

quang-le committed:
IA-3364: fix columns config bug

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: clean up comments

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3364: fix org unit links layout

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
IA-3398: add warning modal when updating phone number
- Update WarningModal o make it reusable
- Add WarningModal if user updates existing phone number

quang-le committed:
IA-3398: update warning modal

- handle all warning cases in one modal

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le committed:
Merge branch 'IA-3364_user_history_front-end' into IA-3398_add_warning_when_updating_phone_number

beygorghor reviewed (changes requested):
- you need to invalid logs while updating a user profile, i had to refresh logs to see changes 
https://github.com/user-attachments/assets/72d17b70-300b-4414-bc8f-37b555f64ed8


- While on a user management or history, this is not reflected in the side menu
Uploading Screen Recording 2024-09-11 at 13.23.45.mov…

- introducing a wrong phone number should show an error above the field
<img width="989" alt="Screenshot 2024-09-11 at 13 24 57" src="https://github.com/user-attachments/assets/924eab63-6e52-4399-8bc3-551cc835bf93">

beygorghor commented:
We can also probably only display when password has been changed also:

<img width="1573" alt="Screenshot 2024-09-11 at 13 32 40" src="https://github.com/user-attachments/assets/0bac5461-4809-47c6-8a6c-28595fa5ae01">

quang-le committed:
IA-3398: add missing translation

quang-le committed:
Merge branch 'IA-3402_add_user_history_api' into IA-3364_user_history_front-end

quang-le commented:
> * you need to invalid logs while updating a user profile, i had to refresh logs to see changes
>       https://github.com/user-attachments/assets/72d17b70-300b-4414-bc8f-37b555f64ed8
> 
>     * While on a user management or history, this is not reflected in the side menu
>       Uploading Screen Recording 2024-09-11 at 13.23.45.mov…
> 
>     * introducing a wrong phone number should show an error above the field
> 
> 
> <img alt="Screenshot 2024-09-11 at 13 24 57" width="989" src="https://private-user-images.githubusercontent.com/12494624/366437488-924eab63-6e52-4399-8bc3-551cc835bf93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNDc1OTIsIm5iZiI6MTcyNjE0NzI5MiwicGF0aCI6Ii8xMjQ5NDYyNC8zNjY0Mzc0ODgtOTI0ZWFiNjMtNmU1Mi00Mzk5LThiYzMtNTUxY2M4MzViZjkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDEzMjEzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxZGFiMmE3M2JjM2I5MDlmMTczYzYyMjA2ODM0YWVjNTJkOTQ5NjNkMWNmMzY5OWY4ZWNiYzI4OWUwNzA0Y2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dzQja14ya-5yemOunjdE4wpkDMSSy8681tMVtjvXtjc">

Showing errors in phone number field is a bit out of scope for this PR in my opinion. Anyway, it would require using the exact same validation list as the backend. I didn't use it in the `PhoneNumber` component, because it's not always reliable (I managed to invalidate a valid belgian number when testing)  and I didn't want to exclude valid phone numbers.

quang-le committed:
IA-3364 reformat comment

quang-le committed:
IA-3364 code review

- update menu and base urtl to highlight current menu location
- update related test plus a hardcoded url in WFP
- invalidate usersHistoryList queryKey when saving a user

quang-le committed:
Merge branch 'IA-3364_user_history_front-end' into IA-3398_add_warning_when_updating_phone_number

quang-le committed:
IA-3364: code review 2
- don't show password updated unless it's True

quang-le committed:
IA-3398: fix phone number warning condition

quang-le committed:
IA-3364: merge IA-3402

quang-le committed:
Merge pull request #1618 from BLSQ/IA-3398_add_warning_when_updating_phone_number

IA-3398 add warning when updating phone number
